plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/ProjectContextMenuTest.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.integration.appengine;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import com.google.cloud.tools.eclipse.appengine.facets.AppEngineStandardFacet;
+import com.google.cloud.tools.eclipse.swtbot.SwtBotProjectActions;
+import com.google.cloud.tools.eclipse.test.util.project.TestProjectCreator;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.jst.common.project.facet.core.JavaFacet;
+import org.eclipse.jst.j2ee.web.project.facet.WebFacetUtils;
+import org.eclipse.swtbot.swt.finder.exceptions.WidgetNotFoundException;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotMenu;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotTreeItem;
+import org.junit.Rule;
+import org.junit.Test;
+
+/** Ensure configured menu options are shown for various types of project. */
+public class ProjectContextMenuTest extends BaseProjectTest {
+  @Rule public TestProjectCreator projectCreator = new TestProjectCreator();
+
+  @Test
+  public void testPlainJavaProject() {
+    IProject project = projectCreator.withFacets(JavaFacet.VERSION_1_8).getProject();
+    SWTBotTreeItem selected = SwtBotProjectActions.selectProject(bot, project.getName());
+    assertContextMenuPathNotFound(selected, "Debug As", "2 App Engine");
```

Comment:
I remember I encountered a problem with the menu number ("2" here) before. It wasn't always the same number depending on environments, so took a different approach in the past. See #2939 for example.

Reply:
Good idea. I started going down that route, but it took a turn towards a cliff and I steered back.  I'll take a look at #2939.

Reply:
Yeah, this code is ugly and can be improved even if by writing a custom matcher.